GChan/ImageLink.cs

```diff
@@ -39,47 +39,55 @@ public string GenerateNewFilename(ImageFileNameFormat format)
-                    break;
-                case ImageFileNameFormat.IDAndOriginalFilename:
-                default:
-                    result = Path.GetFileNameWithoutExtension(URL) + " - " + UploadedFilename + extension;
-                    break;
-            }
+                ImageFileNameFormat.ID => lastPart,
+                ImageFileNameFormat.OriginalFilename => UploadedFilename + extension,
+                ImageFileNameFormat.IDAndOriginalFilename => $"{Path.GetFileNameWithoutExtension(URL)}-{UploadedFilename}{extension}",
+                _ => $"{UploadedFilename}-{Path.GetFileNameWithoutExtension(URL)}{extension}",
+            };
 
-            if (format == ImageFileNameFormat.ID)
-            {
-                result = lastPart;
-            }
-            else if (format == ImageFileNameFormat.OriginalFilename)
-            {
-                result = UploadedFilename + extension;
-            }
-            else if (format == ImageFileNameFormat.IDAndOriginalFilename)
-            {
-                result = Path.GetFileNameWithoutExtension(URL) + " - " + UploadedFilename + extension;
-            }
-            else //ImageFileFormat == OriginalFilenameAndID
-            {
-                result = UploadedFilename + " - " + Path.GetFileNameWithoutExtension(URL) + extension;
-            }
+            return result;
+
+            //-------------------------------------------------
+
+            //switch (format)
+            //{
+            //    case ImageFileNameFormat.ID:
+            //        result = lastPart;
+            //        break;
+            //    case ImageFileNameFormat.OriginalFilename:
+            //        result = UploadedFilename + extension;
+            //        break;
+            //    case ImageFileNameFormat.IDAndOriginalFilename:
+            //    default:
+            //        result = Path.GetFileNameWithoutExtension(URL) + " - " + UploadedFilename + extension;
+            //        break;
+            //}
+
+            //if (format == ImageFileNameFormat.ID)
+            //{
+            //    result = lastPart;
+            //}
+            //else if (format == ImageFileNameFormat.OriginalFilename)
+            //{
+            //    result = UploadedFilename + extension;
+            //}
+            //else if (format == ImageFileNameFormat.IDAndOriginalFilename)
+            //{
+            //    result = Path.GetFileNameWithoutExtension(URL) + " - " + UploadedFilename + extension;
+            //}
+            //else //ImageFileFormat == OriginalFilenameAndID
+            //{
+            //    result = UploadedFilename + " - " + Path.GetFileNameWithoutExtension(URL) + extension;
+            //}
 
             //if (result.Length > FILENAME_MAX_LENGTH)
             //{
             //    result = result.Substring(0, FILENAME_MAX_LENGTH - extension.Length) + extension;
             //}
 
             //Program.Log($"Filename generated: {result}. Length: {result.Length}");

```

Comment:
Remove all of this commented-out-code, apart from the logging on line 90.
The logging on line 90 is outdated too, we have since switched to NLog, would you like to while you are here update this line to use NLog, as is done in other files (refer to them for examples).